src/esthri/src/ops/download.rs

```diff
@@ -0,0 +1,557 @@
+/*
+* Copyright (C) 2021 Swift Navigation Inc.
+* Contact: Swift Navigation <dev@swiftnav.com>
+*
+* This source is subject to the license found in the file 'LICENSE' which must
+* be be distributed together with this source. All other rights reserved.
+*
+* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+*/
+
+use std::fs::File;
```

Comment:
Using stuff from `std::io` and `std::fs` together with async might result in some weird behavior because they have potential to block. I guess we've been doing this all along and haven't really run into problems so maybe not a big deal. It might be because we tend to use the multi-threaded runtime so blocking one thread doesn't block everything. Regardless, using  `tokio:fs`/wrapping the blocking stuff in `spawn_blocking` would probably be a good idea.

Reply:
Wrapped the new blocking stuff in spawn_blocking

Reply:
Wrapped all the blocking writes I could find, moved std::fs/std::io references to an internal module called "bio" so it was obvious if something was interacting with blocking I/O.  Using `tokio::task::spawn_blocking` had the downside of requiring a `'static` lifetime bound, which I've learned (recently) is not the same as `&'static` but this bound ended up propagating to a few odd places (most notably: `upload_from_reader` now requires that you give it ownership of the reader, but this was mostly an internal function anyway).  Might be able to get rid of this bound if we switched to `tokio::fs` but that's probably a bigger change.